### PR TITLE
Large refactor 

### PR DESCRIPTION
**Pull request by GwartneyDev**
Still in the middle of the full refactor, I have the activity progress section complete, I just need to finish calling in the db query methods to finish looping through information.

Just still working my way through the goals file and then need to clean up the main file. 

Take a look the Activity file first to get the best idea of how part of the refactor works. I also renamed a few folders to make more sense with the refactor. Feel free to make suggestions as to what i have done thus far. 

Also you will see a file called helpermethods.go, for now this is where im placing alot of one off functions. I dont want this being a itnerface just yet that has to be implmented each time, as Im not sure if they will need to stay or not.

I also added in some extra custom error loging for our end, so in theory it will create time stamps in certain places you will see the method called something like utils.CreateServerErrorLog("serverLog.txt", timestamp), This calls another method in side the utlils file. Im still working on this as I needed to update another param, and i have this here for now. Will update this part last. 

I should be able to get the rest of this wraped up tomorrow and be able to test all the end points. I just kept having to go back and forth so took me much longer than expected.

### CODE REVIEW

File: backend/.env.example
Comment (benzend) [inline backend/.env.example:1]: why u delete :(

File: backend/Controllers/Activity.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	"github.com/benzend/goalboard/utils"
+	_ "github.com/lib/pq"
+)
+
+var body ActivityProgress
+
+var timestamp = time.Now().Format(time.RFC3339)
+
+// Connect to the database
+var db, err = database.Connect()
```
Comment (benzend) [inline backend/Controllers/Activity.go:21]: i wonder if we should just use db within the context? should be an already existing db reference.

File: backend/Controllers/Activity.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+type ActivityProgress struct {
+	Progress string `json:"Progress"`
```
Comment (benzend) [inline backend/Controllers/Activity.go:24]: lowercase `json:"progress"`

File: backend/Controllers/Activity.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	// Validate the signing method
+	// Return the byte array representation of the secret key
+	// Extract claims from the token
+	// Token is invalid or claims couldn't be extracted
+	claims, shouldReturn := utils.GetUserSessionInfo(sessionInfo, secretRed, w)
+	if shouldReturn {
+		return
+	}
+
+	var userID int
```
Comment (benzend) [inline backend/Controllers/Activity.go:109]: this doesn't seem to be doing anything, do we even need it?

File: backend/Controllers/goals.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	"log"
+	"net/http"
+
+	"github.com/benzend/goalboard/database"
+	"github.com/benzend/goalboard/utils"
+	_ "github.com/lib/pq"
+)
+
+type setGoal struct {
+	Name           string `json:"Name"`
```
Comment (benzend) [inline backend/Controllers/goals.go:16]: lowercase `json:"name"`, same for the others

File: backend/models/dbQueryModels.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package models
+
+//Keeping this up top since its a core part of the api as to how to retreive everything else
+
+// Query to fetch the user ID based on the username obtained from the token
+var getUserQuery = "SELECT id FROM user_ WHERE username = $1;"
```
Comment (benzend) [inline backend/models/dbQueryModels.go:6]: probably capitalize for public access

File: backend/database/migrations/000001_creategoals_table.down.sql
Comment (benzend) [inline backend/database/migrations/000001_creategoals_table.down.sql:1]: no goals_ table drop?

**Review by benzend — APPROVED** [review]
Looking great! Loving the separations of logic.

My only concern is the .env.example removal - curious on why.

File: backend/Controllers/Activity.go (hunk 2 above)
Comment (GwartneyDev) [reply backend/Controllers/Activity.go:24]: @benzend Will do

File: backend/Controllers/Activity.go (hunk 1 above)
Comment (GwartneyDev) [reply backend/Controllers/Activity.go:21]: @benzend Correct I do need to go through and still do this, as I want to add time out, I just have to finish the other half the refactor first. Just didn't quite get this far however will make sure this gets put in.

File: backend/Controllers/Activity.go (hunk 3 above)
Comment (GwartneyDev) [reply backend/Controllers/Activity.go:109]: @benzend Let me double check on this one, it could have been a place hodler I think.

**Review by GwartneyDev — COMMENTED** [review]
@benzend I have gone ahead and reviewed everything going to make the needed changes today, and continue to finish refactor ect.

**Pull request merged**